src/js_schema.hpp

```diff
@@ -218,5 +222,67 @@ realm::Schema Schema<T>::parse_schema(ContextType ctx, ObjectType schema_object,
     return realm::Schema(schema);
 }
 
+template<typename T>
+typename T::Object Schema<T>::object_for_schema(ContextType ctx, const realm::Schema &schema) {
+    ObjectType object = Object::create_array(ctx);
+    uint32_t count = 0;
+    for (auto& object_schema : schema) {
+        Object::set_property(ctx, object, count++, object_for_object_schema(ctx, object_schema));
+    }
+    return object;
+}
+
+template<typename T>
+typename T::Object Schema<T>::object_for_object_schema(ContextType ctx, const ObjectSchema &object_schema) {
+    ObjectType object = Object::create_empty(ctx);
+
+    static const String name_string = "name";
+    Object::set_property(ctx, object, name_string, Value::from_string(ctx, object_schema.name));
+
+    ObjectType properties = Object::create_empty(ctx);
+    for (auto& property : object_schema.properties) {
+        Object::set_property(ctx, properties, property.name, object_for_property(ctx, property));
+    }
+
+    static const String properties_string = "properties";
+    Object::set_property(ctx, object, properties_string, properties);
+
+    static const String primary_key_string = "primaryKey";
+    if (object_schema.primary_key.size()) {
+        Object::set_property(ctx, object, primary_key_string, Value::from_string(ctx, object_schema.primary_key));
+    }
+
+    return object;
+}
+
+template<typename T>
+typename T::Object Schema<T>::object_for_property(ContextType ctx, const Property &property) {
+    ObjectType object = Object::create_empty(ctx);
+
+    static const String name_string = "name";
+    Object::set_property(ctx, object, name_string, Value::from_string(ctx, property.name));
+
+    static const String type_string = "type";
+    const std::string type = property.type != PropertyTypeArray ? string_for_property_type(property.type) : "list";
+    Object::set_property(ctx, object, type_string, Value::from_string(ctx, type));
```

Comment:
I would prefer if the `type` be the `objectType` for `type == "object"`, since we encourage that usage in our documentation to avoid confusion caused by `objectType` meaning two possible things.